tensorflow/python/keras/layers/normalization_test.py

```diff
@@ -213,7 +277,7 @@ def call(self, x, training):
       model = MyModel()
 
       for _ in range(10):
-        x = constant_op.constant(0.5, shape=[1, 1])
+        x = constant_op.constant(0.5, shape=[2, 1])
```

Comment:
Why this change?

Reply:
The reason is... complicated :)

First, Bessel's correction (`N/(N-1)`) is used for the fused implementation but _not_ for the non-fused. There is a discussion of this [here](https://github.com/keras-team/keras/pull/9028).

Then, for the fused case, the [CPU implementation avoids returning NaN when N < 2](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/fused_batch_norm_op.cc#L176-L180), while the GPU implementation (CUDNN) explicitly returns NaN in these cases.

So when the test calls batchnorm with N = 1 and it (with this PR) calls the fused implementation, it returns NaN (on the GPU) and the test fails.
Changing the test size to N = 2 avoids this problem.

This probably isn't ever observed in the real world because batchnorm doesn't work when N is small anyway.

Let me know if you'd like me to take another action here.

Reply:
If we know that N = 1, we should probably not allow calling the fused version on GPU. I'd prefer an error message over silently producing NaNs.

Reply:
Unfortunately this can't easily be done in the Python layer because it only applies in training mode. Adding a check to build() causes a bunch of tests to fail (ones that don't use training mode). It would have to be an in-graph check, which I don't think there's much precedent for(?).

Reply:
You can add a tf.assert in graph check here. This is not unprecedented and no different from adding a check in the kernel itself.

Reply:
I've added an in-graph check using Assert.

Reply:
It caused a bunch of other tests to fail. I'll have to look into them.